File: app/web/src/types/public.ts
export interface PublicCategorySummary {
  slug: string
  name: string
  description: string | null
  icon: string | null
  total_questions: number
  difficulty: string
}

export interface PublicQuizSummary {
  id: number
  title: string
  description: string | null
  question_count: number
  total_attempts: number
  created_at: string
}

export interface PublicHomeResponse {
  featured_categories: PublicCategorySummary[]
  trending_quizzes: PublicQuizSummary[]
}

File: services/api/app/schemas/public.py
from __future__ import annotations

from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel


class PublicCategorySummary(BaseModel):
    slug: str
    name: str
    description: Optional[str]
    icon: Optional[str]
    total_questions: int
    difficulty: str


class PublicQuizSummary(BaseModel):
    id: int
    title: str
    description: Optional[str]
    question_count: int
    total_attempts: int
    created_at: datetime


class PublicHomeResponse(BaseModel):
    featured_categories: List[PublicCategorySummary]
    trending_quizzes: List[PublicQuizSummary]
